[DataGrid] Add API for synchronous validation (#3006)
mui-org/material-ui-x: docs/src/pages/components/data-grid/editing/editing.md

### Client-side validation

> To validate the value in the cells, first add an `preProcessEditCellProps` callback to the [column definition](/api/data-grid/grid-col-def/) of the respective field to validate.

Reviewer: ```suggestion
To validate the value in the cells, first add a `preProcessEditCellProps` callback to the [column definition](/api/data-grid/grid-col-def/) of the field to validate.
```

{{"demo": "pages/components/data-grid/editing/RowEditControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}

### Conditional validation [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)

Having all cells of a row in edit mode allows validating a field based on the value of another one.

> To accomplish that, start by adding the `onEditCellPropsChange` as explained in the [cell editing](#client-side-validation).

Reviewer: ```suggestion
To accomplish that, start by adding the `preProcessEditCellProps` as explained in the [cell editing](#client-side-validation).
```
